### PR TITLE
Chore: add proper release drafts

### LINKED ISSUE #324: Add proper release drafts
We currently have an automatic go releaser (that builds and ships the provider to the TF provider registry) but our releases aren't descriptive. 
We need to have some sort of descriptive changelog for our releases that will comply with the best practices of TF provider and golang repos

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
closes #324 

### Solution

Removed a few settings that are not required.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -64,11 +64,6 @@ changelog:
       - "^.*Update docs.*$"
 release:
   prerelease: auto
-  mode: prepend
```
Comment (TomerHeber) [inline .goreleaser.yml:67]: prepend doesn't make sense. removing.

File: .goreleaser.yml [hunk 2]
```diff
@@ -64,11 +64,6 @@ changelog:
...
-
-  header: |
```
Comment (TomerHeber) [inline .goreleaser.yml:69]: redundant.

File: .goreleaser.yml [hunk 3]
```diff
@@ -64,11 +64,6 @@ changelog:
...
-    ## Release Notes {{ .Tag }} ({{ .Date }})
-
   footer: |
     ## Thanks! the env0 team.
```
Comment (yaronya) [inline .goreleaser.yml:68]: I'd also remove the footer if we're here 😄

Comment (TomerHeber) [reply .goreleaser.yml]: @yaronya - removed.

**Review by yaronya — APPROVED** [review]

**Pull request merged**